src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -56,6 +56,28 @@ def _mkdir_p(self, path, mode=0o755):
             except cephfs.Error as e:
                 raise VolumeException(e.args[0], e.args[1])
 
+    def _get_single_dir_entry(self, dir_path, exclude=[]):
+        """
+        Return a directory entry in a given directory exclusing passed
+        in entries.
+        """
+        try:
+            dir_handle = self.fs.opendir(dir_path)
+        except cephfs.Error as e:
+            raise VolumeError(e.args[0], e.args[1])
+
+        exclude.extend([".", ".."])
+
+        d = self.fs.readdir(dir_handle)
+        d_name = None
+        while d:
+            if not d.d_name in exclude and d.is_dir():
```

Comment:
this should be,
```
if not d.d_name.decode('utf-8') in exclude and d.is_dir():
```
and even more readable would be
```
if d.d_name not in exclude and d.is_dir():
```

d.d_name is a byte string. So in python3 you'll have to explicitly decode it to convert it to a string. Otherwise, I see this in the mgr log after adding some debugging,
```
2019-06-30 18:39:52.033 7f91fccae700 20 mgr[volumes] processing 1 purge job entries
2019-06-30 18:39:52.033 7f91fccae700 20 mgr[volumes] In get_subvolume_trash_entry: trash path is 'b'.''
2019-06-30 18:39:52.033 7f91fccae700 20 mgr[volumes] In pick_purge_dir_from_volume: purge dir is 'b'.''
2019-06-30 18:39:52.033 7f91fccae700 20 mgr[volumes] registering purge job: a.b'.'
2019-06-30 18:39:52.033 7f91fccae700 20 mgr[volumes] In register_job: job for volume a is [(b'.', <PurgeThread(purgejob.1, started daemon 140264988141312)>)]
2019-06-30 18:39:52.033 7f91fd4af700 20 mgr[volumes] processing 1 purge job entries
2019-06-30 18:39:52.034 7f91fd4af700 20 mgr[volumes] In get_subvolume_trash_entry: trash path is 'b'..''
2019-06-30 18:39:52.034 7f91fd4af700 20 mgr[volumes] In pick_purge_dir_from_volume: purge dir is 'b'..''
2019-06-30 18:39:52.034 7f91fd4af700 20 mgr[volumes] registering purge job: a.b'..'
2019-06-30 18:39:52.034 7f91fd4af700 20 mgr[volumes] In register_job: job for volume a is [(b'.', <PurgeThread(purgejob.1, started daemon 140264988141312)>), (b'..', <PurgeThread(purgejob.0, started daemon 140264996534016)>)]
```
Even `.` and `..` get registered trash dirs to be purged.